lib/js_of_ocaml/dune

```diff
@@ -1,7 +1,8 @@
 (library
  (name js_of_ocaml)
  (public_name js_of_ocaml)
- (libraries js_of_ocaml-compiler.runtime)
+ (libraries
+  (re_export js_of_ocaml-compiler.runtime))
```

Comment:
Why this diff ?

Reply:
This is to export the module `Jsoo_runtime` from the js_of_ocaml library, as you suggested https://github.com/ocsigen/js_of_ocaml/pull/1461#discussion_r1882707548

Reply:
I see, I had something different in mind. I think we could introduce a module `Effect_js` in js_of_ocaml the lib (similar to `Sys_js`) that would expose the primitive from jsoo_runtime.